src/main/java/com/nvidia/spark/rapids/jni/ExceptionAtRow.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids.jni;
+
+/**
+ * ExceptionWithRow is an exception thrown by JNI functions when an error
+ * occurs, it indicates the first row which causes an error.
+ * Typically it's used by Ansi mode.
+ * In Spark-Rapids repo, use the following code to get the row value:
+ * `input.getScalarElement(exception.getRowWithError)`
+ *
+ */
+public class ExceptionAtRow extends RuntimeException {
```

Comment:
How about `ExceptionWithRowIndex`? This also needs to modify the C++ struct accordingly (`exception_with_row_index`).